### PR TITLE
Reorganize health service functions, types, and names

### PR DESCRIPTION
**Pull request by tfussell**
Extracted from overly large PR https://github.com/giantswarm/health-service/pull/25

`types.go` should probably just hold type definitions and constructors. Existing constructors were really more like calculations that the service was performing so I moved them into `service.go` and moved common functions that extracted info from a `v1.Node` object into `key` package to reflect how other GS software does this. I also took this opportunity to add a prefix to enumeration values to make their usage more clear.

### CODE REVIEW

File: service/health/service.go [hunk 1]
```diff
@@ -30,17 +34,114 @@ func New(config Config) (*Service, error) {
...
-		Cluster: clusterHealth,
-		Nodes:   nodesHealth,
+		Cluster: clusterStatus,
+		Nodes:   nodeStatus,
 	}
 
 	return &response, nil
 }
+
+func calculateClusterStatus(cluster v1alpha1.StatusCluster, nodes []v1.Node) ClusterStatus {
```
Comment (xh3b4sd) [inline service/health/service.go]: This thing definitely deserves some juicy unit tests.

**Review by xh3b4sd — CHANGES_REQUESTED** [review]

File: service/health/service.go (hunk 1 above)
Comment (tfussell) [reply service/health/service.go]: I've added unit tests for `calculateClusterStatus` and `calculateNodeStatus`. There is 100% test coverage now.

**Review by xh3b4sd — APPROVED** [review]

File: service/health/key/provider.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package key
+
+// Providers is an array of providers which are currently implemented for health checking
+var Providers = [...]string{"aws", "azure", "kvm"}
+
+// IsKnownProvider is a helper function returning true iff the given value is in Providers
```
Comment (TheoBrigitte) [inline service/health/key/provider.go]: Is this a typo ?

```suggestion
// IsKnownProvider is a helper function returning true if the given value is in Providers
```

**Review by TheoBrigitte — APPROVED** [review]
LGTM

File: service/health/key/provider.go (hunk 2 above)
Comment (tfussell) [reply service/health/key/provider.go]: It's something I picked up in my math classes :) `iff` is short for `if and only if`. It's the biconditional meaning `true` will be returned if the value is in `Providers` *and* `false` if the given value is not in `Providers`. This isn't implied if we only use `if`.

**Pull request merged**